docs/react/README.md

```diff
@@ -1,72 +1,186 @@
+`HatchifyProvider` is a component that hosts and provides access to Hatchify-related state. It must be a parent to any Hatchify components.
 
 ```tsx
-const App: React.FC = () => {
+import {HatchifyProvider} from "@hatchifyjs/react"
+
+// Define Schemas
+
+const schemas = { ... }
+
+// Create the Hatched React App instance
+const hatchedReact = hatchifyReact(createJsonapiClient("/api", Schemas))
+
+// Define variables for your Hatchify schema generated components
+const TodoDataGrid = hatchedReact.components.Todo.DataGrid
+
+const HatchedComponent = () => {
   return (
-    <ThemeProvider theme={createTheme()}>
-      <HatchifyProvider>
-        {/* 👀 */}
-        <Everything />
-      </HatchifyProvider>
-    </ThemeProvider>
+    <HatchifyProvider>
+      <TodoDataGrid defaultSort={{ direction: "asc" }} />
+    </HatchifyProvider>
   )
 }
 ```
 
-## hatchifyReact
+### Types
+
+`@hatchifyjs/react` provides a number of [types](./types.md) to assist with app customization.
 
-`hatchifyReact` is a function that initializes the `HatchifyApp` object. Inside of the object you will find `components`, `model`, `state`, and `Everything`. `components`, `model`, and `state`, will be broken down further for each defined schema
+| key                                         | description                                                                             |
+| ------------------------------------------- | --------------------------------------------------------------------------------------- |
+| [`DataGridState`](./types.md#datagridstate) | The return type of the `useDataGridState` hook.                                         |
+| [`CreateType`](./types.md#createtype)       | The type used when data is created.                                                     |
+| [`UpdateType`](./types.md#updatetype)       | Is the type used when data is updated.                                                  |
+| [`HatchifyApp`](./types.md#hatchifyapp)     | the type of the Hatchify App that is generated with [`hatchifyReact()`](#hatchifyreact) |
+| [`RecordType`](./types.md#recordtype)       | Matches a record's schema                                                               |
+
+Learn more about the available types [here](./types.md).
+
+## `hatchedReact`
+
+`hatchedReact` is the instantiated [`HatchifyReact` App](#hatchifyreact) that is returned by the [`hatchifyReact`] constructor function. It provides:
+
+- [Everything](./Everything.md) - is a default component comprised of a set of tabs 🛑 (one for each schema) and DataGrids.
+- [components](./components.md) - a set of components for each of the defined schemas to be used in the app.
+- [model](./model.md) - a set of hooks and promises for each of the defined schemas
+- [state](./state.md) - a set of hooks for each of the defined schemas to be used in the app.
```

Comment:
state - contains hooks that can be used when customizing components provided by hatchify